### PR TITLE
Feature: implement hpilo_info system power info

### LINKED ISSUE #2999: hpilo_info current system power status
### Summary

I don't see in the docs how to get the current `System Power` status.

### Issue Type

Documentation Report

### Component Name

hpilo_info

### Ansible Version

```console (paste below)
ansible 2.10.11
  config file = /code/ansible.cfg
  configured module search path = ['/home/ansible/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.10.0b1 (default, May  4 2021, 17:55:34) [GCC 10.2.1 20201203]
```

### Configuration

```console (paste below)
ANSIBLE_FORCE_COLOR(env: ANSIBLE_FORCE_COLOR) = True
DEFAULT_CALLBACK_WHITELIST(env: ANSIBLE_CALLBACK_WHITELIST) = ['junit', 'profile_tasks']
DEFAULT_GATHERING(env: ANSIBLE_GATHERING) = smart
HOST_KEY_CHECKING(env: ANSIBLE_HOST_KEY_CHECKING) = False
INTERPRETER_PYTHON(env: ANSIBLE_PYTHON_INTERPRETER) = /usr/local/bin/python3
```

### OS / Environment

iLO 4
ProLiant DL360 Gen9

### Additional Information

_No response_

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by YWatchman**
##### SUMMARY
Implement system power state in hpilo_info module.

Fixes #2999 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
hpilo_info

### CODE REVIEW

File: changelogs/fragments/3079-report-power-state-hpilo.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - hpilo_info - report power state of machine with OFF, ON or UNKNOWN
```
Comment (Ajpantuso) [inline changelogs/fragments/3079-report-power-state-hpilo.yaml:3]:
```suggestion
  - hpilo_info - added ``host_power_status`` return value to report power state of machine with OFF, ON or UNKNOWN (https://github.com/ansible-collections/community.general/pull/3079).
```

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 2]
```diff
@@ -113,6 +113,13 @@
     returned: always
     type: str
     sample: 123456ABC78901D2
+
+host_power_status:
+    description: Power status of host
+    returned: always
+    type: str
+    sample: |
+      ON, OFF or UNKNOWN
```
Comment (Ajpantuso) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:122]:
```suggestion
      ON, OFF or UNKNOWN
    version_added: 3.5.0
```

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Just a couple of doc updates.

File: plugins/modules/remote_management/hpilo/hpilo_info.py (hunk 2 above)
Comment (felixfontein) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:122]:
```suggestion
      ON
```
It does only return one of the values, not the string "ON, OFF or UNKNOWN" I think.

File: plugins/modules/remote_management/hpilo/hpilo_info.py [hunk 3]
```diff
@@ -113,6 +113,13 @@
     returned: always
     type: str
     sample: 123456ABC78901D2
+
+host_power_status:
+    description: Power status of host
```
Comment (felixfontein) [inline plugins/modules/remote_management/hpilo/hpilo_info.py:118]:
```suggestion
    description:
      - Power status of host.
      - Will be one of C(ON), C(OFF) and C(UNKNOWN).
```

File: plugins/modules/remote_management/hpilo/hpilo_info.py (hunk 2 above)
Comment (YWatchman) [reply plugins/modules/remote_management/hpilo/hpilo_info.py:122]: The `|` operator in YAML makes it a multi-line string, which I supposed will output this as a string.

Comment (felixfontein) [reply plugins/modules/remote_management/hpilo/hpilo_info.py:124]: `sample` should be an example of a value that is returned by the module. The module does not return the string "ON, OFF or UNKNOWN", but one of the strings "ON", "OFF" or "UNKNOWN". Thus `sample` should be either `ON`, `OFF` or `UNKNOWN`, but **not** the string `ON, OFF or UNKNOWN` as it is right now.

**Review by felixfontein — COMMENTED** [review]
As far as I can tell (I cannot judge the code change itself), it looks good.

**Review by Ajpantuso — APPROVED** [review]
LGTM, thanks!

**Comment by felixfontein** [comment]
@YWatchman I already did review the module after the last changes, see https://github.com/ansible-collections/community.general/pull/3079#pullrequestreview-717694169

**Review by russoz — APPROVED** [review]
LGTM

**Comment by russoz** [comment]
@YWatchman thanks for working on this!
@felixfontein @Ajpantuso thanks for reviewing!

**Pull request merged**